### ISSUE TITLE
Changing maths environment with pmatrix

### ISSUE THREAD

**Issue by thedoctar** [vim-latex/vim-latex#219]
Changing the maths environment to `equation` or `equation*` with a nested `pmatrix` environment will remove the ampersands. Presumably the intention is to remove alignment ampersands, but I would remove this functionality, given that many environments such as `cases` can be used within `equation*` and use alignment ampersands.
```
\begin{equation}
  % Changing environment with Shift+F5 to equation*
  % will delete the first ampersands in the matrix below.
  \begin{pmatrix}
    1 & 0 & 0 \\
    0 & 1 & 0 \\
    0 & 0 & 1
  \end{pmatrix}
\end{equation}
```

**Comment by gerw** [comment]
I agree, the behavior is really questionable. Would it be better to only delete '&' which are not nested inside another environment?

By the way, I found another bug in this function: When going from (e.g.) 'equation*', to 'equation', the function only adds a '\label', if there is a label present after the end of the environment...

**Comment by thedoctar** [comment]
Yes I have also had issues with the label functionality. I would not remove existing labels.

If it's possible to not remove ampersands inside nested environments that's great! I didn't think it would be possible.

**Comment by gerw** [comment]
A hacky way would be to call `Tex_GetCurrentEnv` for each match of the 'delete' pattern within the function `s:Change`. This, however, would be rather slow...

**Comment by thedoctar** [comment]
Is it possible also to avoid the text ampersand `\&`, in case it is used inside `\text{...}`?

There is also the case, maybe rather rare, if you have a nested environment of the same outer one, which is being changed, e.g.
```
\begin{equation}
  \begin{aligned}
    % some maths
    \begin{aligned}
      % Another nested aligned
    \end{aligned}
    % more maths
    % say I want to change the outer aligned
    % using Shift+F5; what happens to the
    % ampersands in the inner aligned?
  \end{aligned}
\end{equation}
```
If you can also detect the nested level, then once you detect that you're in a nested environment, then you can just skip to `\end{env}`. You only have to check the environment once per line as well. How long does it take to call Tex_GetCurrentEnv?
